### ISSUE TITLE
[BUG]: List of feed entries is not read when using a screen reader

### ISSUE THREAD

**Issue by sonor3000** [martinrotter/rssguard#787]
### Brief description of the issue

I am blind and I'd like to use RSS Guard with the orca screen reading software on Linux.

All though not perfect regarding accessebility, RSS Guard is useable for me and thats great! There are not so many accessible RSS reader options for a blind computer user... Thanks for the great work!

The biggest issue  I have with the current RSS Guard release is that the entries of a chosen RSS feed are not read by my screen reading software. The screen reading software is called orca, its outputting the content of a screen with its elements via a synthetic voice or, if a braille device is attached to the computer, also in braille.

I can see the feed entries on my braille device but there is no speech output.

I think its related to the grafical element that is used to generate the entry list, it seems to be another element then the tree or list that is used e.g. for the feeds.

Is there a chance to get the entry list for feed entries also more accessible somehow? That would be really great. I can open also reports for the other accessibility issues, but they are not so central... Because I am no programmer I can't give hints how to improve accessibility unfortunately, but I know there are accessibility guidelines or at least hints how to deal with it.

Thanks a lot for your support and  ofcourse I'd be happy to help at least with testing...

### How to reproduce the bug?

Install orca on a Linux system and use RSS Guard with it. I understand that this very challenging for a sighted person...

### What was the expected result?

The list of feed entries should be spoken by the screen reading software.

### What actually happened?

The ontent of the entry list is not spoken.

### Debug log

I can post a debug log, but I think this will not help in my case...

### Operating system and version

* OS:
4.2.3, manualy installed* RSS Guard version:
Testing

**Comment by schulle4u** [comment]
This is also a problem in Windows builds. E. g. in NVDA you can use object navigation to read the articles list, but this is not very comfortable. I've seen that RSSGuard recently upgraded to Qt6, maybe this will also solve some of the accessibility issues? A dev snapshot for testing would be nice.

**Comment by martinrotter** [comment]
there is, check development build section

**Comment by schulle4u** [comment]
Thanks, found the page. 

@sonor3000 Please try the following workaround:

1. Navigate to the articles list. 
2. Press CTRL+Spacebar to unselect any item. 
3. Press the applications key or Shift+F10 to bring up the columns menu. 
4. Uncheck everything before the title column (e. g. read icon, attachments etc). 

This eventually could make your screen reader correctly announce the articles list, at least in recent development builds. 

Qt6 indeed makes working with screen readers in RSSGuard a lot more stable, however there's still some work to do to make it fully accessible. If I have time this weekend I'll try to compile an issue with the most critical A11Y problems. 

Steffen

**Comment by sonor3000** [comment]
@schulle4u: Thanks for the hint regarding the workaround. I tested it on Win10 with nvda and Jaws, both newest version, but it didn't work. I used the development snapshot of RSS Guard.

I'll test with orca on sunday or monday when I am back hom
e, but if work is done regarding accessibility it would be great if also the Windows versions can be fixed as good as possible.

Ciao and thanks for all your help!

**Comment by martinrotter** [comment]
FOR ME: https://doc.qt.io/qt-6/qt.html#ItemDataRole-enum

"Qt::AccessibleTextRole"
"Qt::AccessibleDescriptionRole"

**Comment by sonor3000** [comment]
@martinrotter and @schulle4u : I tested the sugested workaround also on linux now with the latest dev build for noweb. It did not work. If no feedentry is selected and if I open the context menu I also get the possible to set a label or the second menu entry "open in external program". I do not know how to get into the menu whre I can select which columns shall be displayed in the entry list like I was able todo on Windows. So unfortunately on Linux RSS Guard can not really be used and also on Windows, where the displayed columns can be selected, the screen reading software is not reading anything in the entrylist.

It would be so nice if this can be fixed, because I do not know any other accessible RSS reading software on Windows and Linux which is also able to work with centralized RSS aggregators like TT-RSS or Miniflux.

Thanks a lot for your work and support!

**Comment by martinrotter** [comment]
Guys.

I made some tests with NVDA and found out what @schulle4u wrote above. It just works, all you need is to hide columns "Read" and "Starred",  after it the list is read by NVDA without problems. Rest of app seems to be read good.

**Comment by martinrotter** [comment]
Closing as of now I tested the "workaround" which seems to work.
